discovery_engine_core/ai/xayn-ai/src/coi/key_phrase.rs

```diff
@@ -95,21 +105,46 @@ pub(crate) struct KeyPhrases {
     removed: HashMap<CoiId, Vec<KeyPhrase>>,
 }
 
+/// Smart pointer to candidates' data.
+#[derive(Clone, Copy)]
+pub(crate) struct Candidates<'c, S> {
+    candidates: &'c [String],
+    market: &'c Market,
+    smbert: S,
+}
+
+impl<'c, S> Candidates<'c, S> {
+    pub(crate) fn new(candidates: &'c [String], market: &'c Market, smbert: S) -> Option<Self> {
```

Comment:
:grey_question: Can you please explain more about the decision of having `Option<Self>` instead of allowing an empty `candidates` fields? At the moment I don't think it adds much value.

Reply:
please see https://github.com/xaynetwork/xayn_discovery_engine/pull/302#discussion_r848350323